src/analysis/convex/cone/proper.lean

```diff
@@ -46,8 +45,106 @@ protected def closure (K : convex_cone ℝ E) : convex_cone ℝ E :=
 
 @[simp, norm_cast] lemma coe_closure (K : convex_cone ℝ E) : (K.closure : set E) = closure K := rfl
 
-protected lemma mem_closure {K : convex_cone ℝ E} {a : E} :
-  a ∈ K.closure ↔ a ∈ closure (K : set E) :=
-iff.rfl
+@[simp] protected lemma mem_closure {K : convex_cone ℝ E} {a : E} :
+  a ∈ K.closure ↔ a ∈ closure (K : set E) := iff.rfl
+
+lemma closure_eq_iff_is_closed {K : convex_cone ℝ E} : K.closure = K ↔ is_closed (K : set E) :=
+⟨ (λ h, by rw [← closure_eq_iff_is_closed, ← coe_closure, h]),
+  (λ h, set_like.coe_injective $ closure_eq_iff_is_closed.2 h) ⟩
 
 end convex_cone
+
+section definitions
+
+/-- A proper cone is a convex cone `K` that is nonempty and closed. Proper cones have the nice
+property that the dual of the dual of a proper cone is itself. This makes them useful for defining
+cone programs and proving duality theorems. -/
+structure proper_cone (E : Type*) [normed_add_comm_group E] [inner_product_space ℝ E]
+  extends convex_cone ℝ E :=
+(nonempty'  : (carrier : set E).nonempty)
+(is_closed' : is_closed (carrier : set E))
```

Comment:
Maybe a question more for Eric: do you think there should be a `convex_cone_class` that extends `set_like`?

Reply:
If it's ok, I can work on this in a separate PR. There was another suggestion to bundle the maps in `convex_cone` and `proper_cone`. I can work on adding both of these once the Farkas' lemma proof is done.